### PR TITLE
GUACAMOLE-525: Fix issues with SQL Server compatibility

### PR DESCRIPTION
**Pull request by necouchman**
Fixes a couple of issues reported with SQL Server compatibility and the lack of support for the LIMIT keyword.

### CODE REVIEW

File: extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml [hunk 1]
```diff
@@ -66,12 +65,11 @@
         );
 
         DELETE FROM [guacamole_user_password_history]
-        WHERE password_history_id IN (
-            SELECT password_history_id
+        WHERE password_history_id NOT IN (
```
Comment (mike-jumper) [inline extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml]: Won't this now delete password history for all other users?

Comment (mike-jumper) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml]: I think I see what you're trying to do here now. Maybe needs an `AND user_id = #{record.userID,jdbcType=INTEGER}` at the level of the outer select?

Comment (necouchman) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml]: Ummm...yes.  Probably not the behavior we want.

Comment (necouchman) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml]: > I think I see what you're trying to do here now. Maybe needs an AND user_id = #{record.userID,jdbcType=INTEGER} at the level of the outer select?

Yep!  Commit coming right up...

Comment (necouchman) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml]: Should be fixed.

**Comment by necouchman** [comment]
@mike-jumper Anything else that needs fixing here?

**Comment by mike-jumper** [comment]
Looks good to me. Will merge shortly.

**Pull request merged**